app/controllers/reports_controller.rb

```diff
@@ -11,22 +11,25 @@ def index
   def show
     # are we searching for the last report?
     if params[:id] == "last"
-      conditions = { :host_id => Host.find(params[:host_id]).try(:id) } unless params[:host_id].blank?
+      conditions = { :host_id => Host.authorized.find(params[:host_id]).try(:id) } unless params[:host_id].blank?
```

Comment:
Use `.authorized(:view_hosts)` rather than plain .authorized, otherwise I think *any* host permission would be sufficient.